### PR TITLE
Add manual_saturating_arithmetic lint

### LINKED ISSUE #1557: Lint to suggest `.saturating_add/sub(x)` for `.checked_add/sub(x).unwrap_or(MAX/MIN)`
```rust
unsigned.checked_add(x).unwrap_or(unsigned::MAX)
 -> unsigned.saturating_add(x)
unsigned.checked_sub(x).unwrap_or(0)
 -> unsigned.saturating_sub(x)
```

Signed versions can also be linted if literals are used.

```
signed.checked_add(non_negative_literal).unwrap_or(signed::MAX)
 -> signed.saturating_add(x)
signed.checked_add(    negative_literal).unwrap_or(signed::MIN)
 -> signed.saturating_add(x)

signed.checked_sub(non_negative_literal).unwrap_or(signed::MIN)
 -> signed.saturating_sub(x)
signed.checked_sub(    negative_literal).unwrap_or(signed::MAX)
 -> signed.saturating_sub(x)
```

### PR DESCRIPTION
**Pull request by sinkuu**
changelog: add `manual_saturating_arithmetic` lint

Fixes #1557. This lint detects manual saturating arithmetics like `x.checked_add(10u32).unwrap_or(u32::max_value())` and suggests replacing with `x.saturating_add(10u32)`.

### CODE REVIEW

File: clippy_lints/src/methods/mod.rs [hunk 1]
```diff
@@ -983,6 +984,31 @@ declare_clippy_lint! {
...
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for `.checked_add/sub(x).unwrap_or(MAX/MIN)`.
+    ///
+    /// **Why is this bad?** These can be written simply with `saturating_add/sub` methods.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// let x: u32 = 100;
```
Comment (flip1995) [inline clippy_lints/src/methods/mod.rs]:
```suggestion
    /// # let y: u32 = 0;
    /// # let x: u32 = 100;
```

File: clippy_lints/src/methods/mod.rs [hunk 2]
```diff
@@ -983,6 +984,31 @@ declare_clippy_lint! {
...
+    ///
+    /// let add = x.checked_add(y).unwrap_or(u32::max_value());
+    /// let sub = x.checked_sub(y).unwrap_or(u32::min_value());
+    /// ```
+    ///
+    /// can be written using dedicated methods for saturating addition/subtraction as:
+    ///
+    /// ```rust
```
Comment (flip1995) [inline clippy_lints/src/methods/mod.rs]: Same as above.

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Comment by lzutao** [comment]
I suggest using `syntax::ast::LitIntType` in `syntax::ast::LitKind::Int`. This would make the code clearer.

**Comment by sinkuu** [comment]
@Izutao There's `LitIntType::Unsuffixed` variant, so I'd need to rely on type information anyway.

**Pull request merged**